IKZ_plugin/src/ikz_plugin/__init__.py

```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+        """,
+        section_def=Component,
+        repeats=True,
+    )
+
+
+class QuantifySolidMaterial(QuantifyMaterial, SolutionPreparationStep):
+    """
+    Class autogenerated from yaml schema.
+    """
+
+    m_def = Section(
+        a_eln=None,
+        categories=[IKZCategory],
+    )
+
+
+class QuantifyLiquidMaterial(QuantifyMaterial, SolutionPreparationStep):
+    """
+    Class autogenerated from yaml schema.
+    """
+
+    m_def = Section(
+        a_eln=None,
+        categories=[IKZCategory],
+    )
+    measured_volume = Quantity(
+        type=float,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "liter"},
+        unit="liter",
+    )
+    density = Quantity(
+        type=float,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "gram / liter"},
+        unit="gram / liter",
+    )
+
+
+class MixMaterial(Process, SolutionPreparationStep):
+    """
+    Mix the quantified materials.
+    """
+
+    m_def = Section(
+        a_eln=None,
+        categories=[IKZCategory],
+    )
+    mixing_time = Quantity(
+        type=float,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "second"},
+        unit="second",
+    )
+    temperature = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "celsius"},
+        unit="celsius",
+    )
+    container_type = Quantity(
+        type=str,
+        a_eln={"component": "StringEditQuantity"},
+    )
+    rotation_speed = Quantity(
+        type=float,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "rpm"},
+        unit="rpm",
+    )
+    mixed_material = SubSection(
+        description="""
+        The mixed material.
+        """,
+        section_def=CompositeSystem,
+    )
+
+
+class SolutionPreparationIKZ(Process, EntryData):
+    """
+    Solution preparation class
+    """
+
+    method = Quantity(
+        type=MEnum("Shaker", "Ultrasoncic", "Waiting", "Stirring"),
+        shape=[],
+        a_eln=dict(
+            component="EnumEditQuantity",
+        ),
+    )
+    description = Quantity(
+        type=str,
+        a_eln={"component": "StringEditQuantity"},
+    )
+    atmosphere = Quantity(
+        type=str,
+        a_eln={"component": "StringEditQuantity"},
+    )
+    intended_sample_conc = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
+        unit="mol / liter",
+    )
+    obtained_conc = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
```

Comment:
label

Reply:
I moved this in the mixing. I think it is fine to have it there. Also, I made it as an array.

But I'm open to delete this quantity, too. It might be included at the single use case level, if needed

Reply:
moved it in ComponentConcentration class